### DISCUSSION TITLE
How can I build an outputed file with all depencies ?

### DISCUSSION THREAD

**Discussion by Suniron** [egoist/tsup#614] [Q&A]
Hi,

I think that my question is about a basic usage but I can't find instruction in documentation or other issues and discussions..

So, I want to build a single **index.js** wich inclue all other typescript files (it's already good for that) and all depencies, on a typescript project.

Actually, if i do a `tsup src/index.ts` at root, a dist folder is built with the famous **index.js** inside.
But, this **index.js** work only in this **dist** folder (or at root of project). 

So, if I put this file on my production server (without any **package.json** or **node_modules** folder), and I do a `node index.js`, some error are printed to say that modules aren't found (**package.json** depencies).

Could you help me to find the correct arguments to include all depencies at build time 🙏.

Thanks by advance.

**Comment by huangzl97** [comment]
[by default tsup will exclude `dependencies` and `peerDependencies`](https://tsup.egoist.sh/#excluding-packages)


my tsup version is `6.1.2`, and i found that there is a new option `noExternal` which @egoist  didn't mention on the documentation website but [typedoc](https://paka.dev/npm/tsup@6.1.2/api#9286de5cf1f8699). so i have found a solution base on `noExternal` and it works fine for me:
```ts
// your tsup.config.ts
import { dependencies, devDependencies } from './package.json';
export default defineConfig({
  noExternal: Object.keys(dependencies).concat(Object.keys(devDependencies))
})
```
